model-optimizer/extensions/front/tf/if_ext.py

```diff
@@ -0,0 +1,113 @@
+    then_graph = Graph()
+    # fill the body graph
+    for attr_key in main_graph.graph.keys():
+        if attr_key != 'library':
+            then_graph.graph[attr_key] = copy.deepcopy(main_graph.graph[attr_key])
+        else:
+            # it is sufficient to have a link to the library
+            then_graph.graph['library'] = main_graph.graph['library']
+    if_node['then_graph'] = then_graph
+
+    # TODO refactor to have one function doing this
+    # create "else" graph
+    else_graph = Graph()
+    # fill the body graph
+    for attr_key in main_graph.graph.keys():
+        if attr_key != 'library':
+            else_graph.graph[attr_key] = copy.deepcopy(main_graph.graph[attr_key])
+        else:
+            # it is sufficient to have a link to the library
+            else_graph.graph['library'] = main_graph.graph['library']
+    if_node['else_graph'] = else_graph
+
+    # create Parameter nodes for the then/else graphs
+    for input_index, (body_graph, body_graph_proto) in enumerate(zip((then_graph, else_graph), (then_graph_proto,
+                                                                                                else_graph_proto))):
+        body_parameters = []
+        body_parameter_names = []
+        for idx, pb_node in enumerate(body_graph_proto['input_arg']):
+            param_id = body_graph.unique_id(pb_node.name)
+            body_graph.add_node(param_id, name=param_id, kind='op', op='Parameter', pb=None, shape=None)
+            parameter_node = Node(body_graph, pb_node.name)
+            Parameter.update_node_stat(parameter_node,
+                                       {'data_type': tf_dtype_extractor(pb_node.type),
+                                        'permute_attrs': PermuteAttrs().update_attrs(attrs=[('shape', 'output:0')])}
+                                       )
+            body_parameters.append(parameter_node)
+            body_parameter_names.append(param_id)
+
+        # update the If body graph with the body function graph
+        body_results = []
+        update_body_graph(body_graph, body_graph_proto, body_parameter_names, body_results)
+
+        body_graph.stage = 'front'
+
+        # connect external input ports with body parameter nodes except input with condition
+        for idx in range(0, len(body_parameters)):
+            If.connect_body_input(if_node, not input_index, idx + 1, body_parameters[idx])
+
+        # connect body outputs with If operation output ports
+        for idx in range(len(body_results)):
+            If.connect_body_output(if_node, not input_index, idx, body_results[idx])
+
+        # # run function to parse body nodes attributes similar to the main graph
+        extract_node_attrs(body_graph, lambda node: tf_op_extractor(node, check_for_duplicates(tf_op_extractors)))
+    return cls.enabled
+
+
+class IfExtractor(FrontExtractorOp):
+    op = 'If'
+    enabled = True
+
+    @classmethod
+    def extract(cls, if_node: Node):
```

Comment:
Maybe we need also extractor for ONNX If?

Reply:
no, we need if now for enabling tf2 center_net models. Full support if in next release